csrc/mmdeploy/backend_ops/tensorrt/modulated_deform_conv_v3/trt_deform_conv_v3.cpp

```diff
@@ -0,0 +1,297 @@
+  deserialize_value(&data, &length, &pad_h_);
+  deserialize_value(&data, &length, &pad_w_);
+  deserialize_value(&data, &length, &dilation_h_);
+  deserialize_value(&data, &length, &dilation_w_);
+  deserialize_value(&data, &length, &group_);
+  deserialize_value(&data, &length, &group_channels_);
+  deserialize_value(&data, &length, &offset_scale_);
+  deserialize_value(&data, &length, &im2col_step_);
+}
+
+nvinfer1::IPluginV2DynamicExt *TRTDCNv3::clone() const TRT_NOEXCEPT {
+  TRTDCNv3 *plugin =
+      new TRTDCNv3(mLayerName, kernel_h_, kernel_w_, stride_h_, stride_w_, pad_h_, pad_w_,
+                   dilation_h_, dilation_w_, group_, group_channels_, offset_scale_, im2col_step_);
+  plugin->setPluginNamespace(getPluginNamespace());
+
+  return plugin;
+}
+
+const nvinfer1::IDimensionExpr *output_size(const nvinfer1::IDimensionExpr &input, int pad,
+                                            int dilation, int kernel, int stride,
+                                            nvinfer1::IExprBuilder &exprBuilder) {
+  // out_expand = 2×padding[0]−dilation[0]×(kernel_size[0]−1)+1
+  auto out_expand = exprBuilder.constant(2 * pad - dilation * (kernel - 1) + 1);
+  // out = out_expand + input
+  auto out_before_div = exprBuilder.operation(DimensionOperation::kSUM, input, *out_expand);
+  // out = out / stride
+  auto out_before_sub = exprBuilder.operation(DimensionOperation::kFLOOR_DIV, *out_before_div,
+                                              *(exprBuilder.constant(stride)));
+  // out -=1
+  auto out =
+      exprBuilder.operation(DimensionOperation::kSUB, *out_before_sub, *(exprBuilder.constant(1)));
+  return out;
+}
+
+nvinfer1::DimsExprs TRTDCNv3::getOutputDimensions(
+    int outputIndex, const nvinfer1::DimsExprs *inputs, int nbInputs,
+    nvinfer1::IExprBuilder &exprBuilder) TRT_NOEXCEPT {
+  nvinfer1::DimsExprs ret;
+  ret.nbDims = 4;
+  ret.d[0] = inputs[0].d[0];
+  ret.d[3] = exprBuilder.constant(group_ * group_channels_);
+
+  ret.d[1] = output_size(*inputs[0].d[1], pad_h_, dilation_h_, kernel_h_, stride_h_, exprBuilder);
+  ret.d[2] = output_size(*inputs[0].d[2], pad_w_, dilation_w_, kernel_w_, stride_w_, exprBuilder);
+
+  return ret;
+}
+
+bool TRTDCNv3::supportsFormatCombination(int pos, const nvinfer1::PluginTensorDesc *ioDesc,
+                                         int nbInputs, int nbOutputs) TRT_NOEXCEPT {
+  if (pos == 0) {
+    return (ioDesc[pos].type == nvinfer1::DataType::kFLOAT &&
+            ioDesc[pos].format == nvinfer1::TensorFormat::kLINEAR);
+
+  } else {
+    return ioDesc[pos].type == ioDesc[0].type && ioDesc[pos].format == ioDesc[0].format;
+  }
+}
+
+void TRTDCNv3::configurePlugin(const nvinfer1::DynamicPluginTensorDesc *inputs, int nbInputs,
+                               const nvinfer1::DynamicPluginTensorDesc *outputs,
+                               int nbOutputs) TRT_NOEXCEPT {}
+
+size_t TRTDCNv3::getWorkspaceSize(const nvinfer1::PluginTensorDesc *inputs, int nbInputs,
```

Comment:
Since you have not use extra workspace in the kernel. I think workspace size should be 0.